research/picai/prepare_data.py

```diff
@@ -0,0 +1,209 @@
+import argparse
+import json
+import os
+from pathlib import Path
+from typing import Optional, Sequence, Tuple
+
+import numpy as np
+import SimpleITK as sitk
+from preprocessing import (
+    AlignOriginAndDirection,
+    BinarizeAnnotation,
+    CentreCropAndOrPad,
+    PicaiCase,
+    PreprocessingSettings,
+    Resample,
+    ResampleToFirstScan,
+    preprocess,
+)
+
+DEFAULT_TRANSFORMS = [
+    Resample(),
+    CentreCropAndOrPad(),
+    ResampleToFirstScan(),
+    AlignOriginAndDirection(),
+    BinarizeAnnotation(),
+]
+
+
+def get_labels(paths_for_each_sample: Sequence[Tuple[Sequence[Path], Path]]) -> Sequence[float]:
+    _, annotation_paths = zip(*paths_for_each_sample)
+    case_labels = []
+    for path in annotation_paths:
+        annotation = sitk.ReadImage(path)
+        annotation_array = sitk.GetArrayFromImage(annotation)
+        case_label = float(np.max(annotation_array))
+        case_labels.append(case_label)
+    return case_labels
+
+
+def generate_dataset_json(
+    paths_for_each_sample: Sequence[Tuple[Sequence[Path], Path]],
+    write_dir: Path,
+    splits_path: Optional[Path] = None,
+) -> None:
+
+    labels = get_labels(paths_for_each_sample)
+
+    # Since we are storing paths inside a json file, we need to convert to strings to be serializable.
+    # Also pack in label for each samples in the tuple containing scan paths and annotation path.
+    scan_annotation_label_list = [
+        (list(map(str, tup[0])), str(tup[1]), lbl) for tup, lbl in zip(paths_for_each_sample, labels)
```

Comment:
You should be able to unpack the tuple into labels with more descriptive names as
`... for (name_1, name_2), label`
I'd also expand lbl to label for clarity.